### PR TITLE
fix(browserify/test): Don't let `yarn link` clobber global directory

### LINKED ISSUE #682: browserify tests attempt to modify system environment
As can be observed [here](https://github.com/LavaMoat/LavaMoat/actions/runs/5996407503/job/16260892432?pr=681#step:10:171) (#681), `yarn link` invocation in `lavamoat-browserify` tests can trigger attempts at linking global bins.

Fix: Use temporary directory instead, as already done with projectDir.

### PR DESCRIPTION
**Pull request by legobeat**
Fixes: #682 

Unblocks: #681

### CODE REVIEW

**Comment by legobeat** [comment]
Seems like this is not a complete fix after all 
https://github.com/LavaMoat/LavaMoat/actions/runs/5997272802/job/16263345149#step:10:475

File: packages/browserify/test/fixtures/secureBundling/lavamoat/node/policy.json [hunk 1]
```diff
@@ -60,14 +60,10 @@
     "@lavamoat/aa": {
       "builtin": {
         "fs.readFileSync": true,
-        "fs.statSync": true,
         "path.dirname": true,
         "path.join": true,
         "path.relative": true
       },
-      "globals": {
-        "performantResolve": true
```
Comment (naugtur) [inline packages/browserify/test/fixtures/secureBundling/lavamoat/node/policy.json]: Do you know why this change is here?

**Review by naugtur — APPROVED** [review]

File: packages/browserify/test/fixtures/secureBundling/lavamoat/node/policy.json (hunk 1 above)
Comment (legobeat) [reply packages/browserify/test/fixtures/secureBundling/lavamoat/node/policy.json]: No, I also raised my eyebrows on it. Omit it?

Comment (boneskull) [reply packages/browserify/test/fixtures/secureBundling/lavamoat/node/policy.json]: It looks correct to me.  In fact, running `npm run test:prep` in `main` results in this change.

I think we need to add something to ensure these generated policy files get updated properly, but I'm not sure what that looks like yet.  I'll make an issue for it.

**Review by boneskull — APPROVED** [review]

**Comment by boneskull** [comment]
See https://github.com/LavaMoat/LavaMoat/issues/686

File: packages/browserify/test/fixtures/secureBundling/lavamoat/node/policy.json (hunk 1 above)
Comment (legobeat) [reply packages/browserify/test/fixtures/secureBundling/lavamoat/node/policy.json]: Waiting for merge of #684 (which includes this change) to get this commit clean from unrelated changes.

**Comment by boneskull** [comment]
@legobeat _ping_

**Comment by legobeat** [comment]
- Superseded by #698 + #625
